lib/Conversion/TTIRToTTNN/TTIRToTTNN.cpp

```diff
@@ -542,6 +542,41 @@ class MaxPool2dOpConversionPattern
   }
 };
 
+class BroadcastOpConversionPattern
+    : public OpConversionPattern<ttir::BroadcastOp> {
+  using OpConversionPattern<ttir::BroadcastOp>::OpConversionPattern;
+
+public:
+  LogicalResult
+  matchAndRewrite(ttir::BroadcastOp srcOp, ttir::BroadcastOp::Adaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+
+    mlir::Value input = srcOp.getOperand(0);
+    mlir::Value result = srcOp.getResult();
+    
+    // Try to fold this operation into all Element wise consumer ops.
+
+    for (Operation *nextOp : srcOp->getUsers()) {
+      if (nextOp == srcOp) {
```

Comment:
What is this check for? I

Reply:
This is to get all Uses (consumers) of this Value and to ignore the current Op from the list of Uses.

Reply:
yeah, I guess I meant why would the current op be in the list of consumers?

Reply:
That's a good catch, the list was being mutated while iterated, needed to make a copy first. Updated with change.